android/app/src/main/java/com/sample/react/TestAppReactNativeHost.kt

```diff
@@ -15,21 +16,33 @@ class TestAppReactNativeHost @Inject constructor(
         private val reactBundleNameProvider: ReactBundleNameProvider
 ) : ReactNativeHost(application) {
 
-    var currentActivity: ReactActivity? = null
+    private var currentActivity: Activity? = null
```

Comment:
@ArazAbishov: Looks like I can't get rid of this. The Back button won't behave correctly if we don't provide the correct activity on reload, e.g. it will close the app if you press back from a `ComponentActivity`.

Reply:
Okay, just keep it here for now. Could you please create a ticket, so I can take a look at it later?

Reply:
All yours: #66